packages/app-mining/src/components/WithdrawModal.tsx

```diff
@@ -17,13 +19,20 @@ import type {StakePoolModalProps} from './StakePoolTable'
 
 const WithdrawModal = (props: StakePoolModalProps): JSX.Element => {
   const [polkadotAccount] = usePolkadotAccountAtom()
+  const polkadotAccountAddress = polkadotAccount?.address
   const {onClose, stakePool} = props
   const {api} = useApiPromise()
   const waitSignAndSend = useWaitSignAndSend()
   const decimals = useDecimalJsTokenDecimalMultiplier(api)
   const [amount, setAmount] = useState<number | undefined>()
   const {refetch} = useSelfUserStakeInfo(stakePool.pid)
 
+  const polkadotAccountBalanceDecimal = usePolkadotAccountBalanceDecimal(
```

Comment:
Looks like you are reading the account balance. It's not correct. We should read from `api.query.phalaStakePool.poolStakers([pid, account]).unwrap().shares`.